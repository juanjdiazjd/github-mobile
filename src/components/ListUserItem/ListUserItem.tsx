import * as React from 'react';
import { Alert, Image, Text, View, ViewProps } from 'react-native';
import { Button } from 'react-native-paper';
import styled from 'styled-components';
import theme from '../../theme';
import { UserItem } from '../../types/responseType';

const ListContainer = styled(View)`
  background-color: ${theme.colors.white};
  height: auto;
  width: 350px;
  padding: 15px;
  margin-bottom: 10px;
  border-radius: 10px;
  flex-direction: row;
  justify-content: space-between;
`;

const ComponentContainer = styled(View)`
  flex-direction: row;
  justify-content: center;
  height: auto;
  width: auto;
`;

export const TextCustom = styled(Text)<{
  color?: string;
}>`
  color: ${({ color }) => color ?? theme.colors.titleText};
  top: 10%;
  padding-left: 20px;
  font-size: 14px;
`;

export const TextTitle = styled(Text)<{
  color?: string;
}>`
  color: ${({ color }) => color ?? theme.colors.titleText};
  top: 10%;
  flex: 1;
  flex-wrap: wrap;
  padding-left: 20px;
  font-size: 14px;
`;

const Column = styled(View)`
  flex-direction: column;
`;

const Row = styled(View)`
  flex-direction: row;
`;
const Avatar = styled(Image)`
  width: 40px;
  height: 40px;
  border-radius: 200px;
`;

interface ListUserItemViewProps extends ViewProps {
  item: UserItem;
}

export const ListUserItem: React.FunctionComponent<ListUserItemViewProps> = ({
  item,
}) => {
  return (
    <ComponentContainer>
      {item && (
        <ListContainer>
          <Row>
            <Column>
              <Avatar
                source={{
                  uri: item.avatar_url,
                }}
              />
            </Column>
            <TextCustom color={theme.colors.iris}>{item.login}</TextCustom>
          </Row>
          <Button onPress={() => Alert.alert('Followed')} mode="outlined">
            Follow
          </Button>
        </ListContainer>
      )}
    </ComponentContainer>
  );
};

export default ListUserItem;
