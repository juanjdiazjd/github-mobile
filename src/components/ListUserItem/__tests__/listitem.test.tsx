import React from 'react';
import renderer from 'react-test-renderer';
import { UserItem } from '../../../types/responseType';
import ListUserItem from '../ListUserItem';
jest.mock('swr');
jest.mock('react-native-paper');
jest.mock('react-native-localization');

jest.resetAllMocks();
jest.useFakeTimers();

const MOCK_DATA_LIST_USER_ITEM: UserItem = {
  login: 'pepito',
  id: 131574,
  node_id: 'MDQ6VXNlcjEzMTU3NA==',
  avatar_url: 'https://avatars.githubusercontent.com/u/131574?v=4',
  gravatar_id: '',
  url: 'https://api.github.com/users/pepito',
  html_url: 'https://github.com/pepito',
  followers_url: 'https://api.github.com/users/pepito/followers',
  following_url: 'https://api.github.com/users/pepito/following{/other_user}',
  gists_url: 'https://api.github.com/users/pepito/gists{/gist_id}',
  starred_url: 'https://api.github.com/users/pepito/starred{/owner}{/repo}',
  subscriptions_url: 'https://api.github.com/users/pepito/subscriptions',
  organizations_url: 'https://api.github.com/users/pepito/orgs',
  repos_url: 'https://api.github.com/users/pepito/repos',
  events_url: 'https://api.github.com/users/pepito/events{/privacy}',
  received_events_url: 'https://api.github.com/users/pepito/received_events',
  type: 'User',
  site_admin: false,
  score: 1,
};

const ListItemComponent = (): JSX.Element => (
  <ListUserItem item={MOCK_DATA_LIST_USER_ITEM} />
);

it('Should list user item component', () => {
  const tree = renderer.create(ListItemComponent()).toJSON();
  expect(tree).toMatchSnapshot();
});
