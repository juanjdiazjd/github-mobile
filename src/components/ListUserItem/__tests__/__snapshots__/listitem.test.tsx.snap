// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should list user item component 1`] = `
<View
  className="sc-gsnTZi fluHiX"
>
  <View
    className="sc-bczRLJ cBAjUe"
  >
    <View
      className="sc-jSMfEi hhyYpf"
    >
      <View
        className="sc-eCYdqJ brkrER"
      >
        <Image
          className="sc-gKXOVf lovrYZ"
          source={
            Object {
              "uri": "https://avatars.githubusercontent.com/u/131574?v=4",
            }
          }
        />
      </View>
      <Text
        className="sc-dkzDqf hEKvnl"
        color="#3D52D5"
      >
        pepito
      </Text>
    </View>
  </View>
</View>
`;
